### DISCUSSION TITLE
[Layout] Layout children with params not handling refresh

### DISCUSSION THREAD

**Discussion by cmoynard** [expo/router#256] [Q&A]
I'm doing a very simple site of drinks with this architecture :
```
-| app
--| drinks
---| (editDrink)
---- [drinkId].tsx *
--- index.tsx *
-- _layout.tsx
-- index.tsx *
```

I'm having some trouble with the `_layout` in `drinks` folder. At first it's applied to all the children (* in the folder order)

<img width="553" alt="image" src="https://user-images.githubusercontent.com/78808413/216992199-d830bb7b-b321-4041-a37e-d5f6a3bab92d.png">

In the index, we have a list of the drinks and we can click one to see the details, with the drinks.id in the params of `[drinkId].tsx`

<img width="304" alt="image" src="https://user-images.githubusercontent.com/78808413/216989745-b260a14c-d439-40af-bfad-a8e0959b2cd4.png">

But when i get linked to drinks/4 for example, all of the layout design is gone, all the button are working fine but no more icons, the paddings and margins are going crazy. And it's going on even after i navigate back to others pages, until i refresh outside `drinks/[drinkId]`

https://user-images.githubusercontent.com/78808413/216996372-a19526aa-f840-4212-8bbe-58d17dfefd29.mov

If someone as already got this type of issue or if you need further informations, please let me know and i will provide you the most informations i can !

**Comment by EvanBacon** [comment]
That doesn't seem related to Expo Router. What console warnings/errors are you getting? Also Metro doesn't currently support Suspense, when we add support it'll be wrapped around the routes automatically.

**Reply by cmoynard** [reply]
For the suspense, i cannot remove it right now, because it doesn't support without `<Suspense>`

However I've got several errors when i navigate to `drinks/[drinkId]` : 

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/78808413/217255625-e36e5e4a-64d3-456f-bf66-c9bf27787a13.png">

`Warning : React.dom` and `setNativeProps` are persistent errors and warnings, so i think this errors list is the problem  : 

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/78808413/217256176-a09a54ff-e1c7-43bf-b80d-42ea294abcc3.png">

(Also, the `drinks/[drinkId]` is the only page that refresh on navigation, the others pages are not refreshing when i navigate so it's definitely a "on refresh" problem)

**Comment by cmoynard** [comment]
So I did some tests and find out the `<Suspense>` is not the real issue here, because I redesigned all of my hooks to bypass the `<Suspense>` and it's the same problem, layout gives up on `drinks/[drinkId]` refresh.

But I did find another discussion with the same problem #274  about icons being not handled in expo/router, especially with `@expo/vector-icons`. Maybe the lib is just not supported by expo/router right now.

**Reply by cmoynard** [reply]
I replaced all of my `@expo/vector-icons` with some icons from NativeBase and it works fine

<img width="1044" alt="image" src="https://user-images.githubusercontent.com/78808413/218992246-d22f1423-0198-4f01-8deb-c308d7abfe66.png">

So it's definitively a icon problem, with a better topic right here 👉 #274
